### ISSUE TITLE
train with custom datasets

### ISSUE THREAD

**Issue by kaikaizhu** [sfzhang15/ATSS#22]
When I trained with my own dataset, an error occurred.
I refer to question 14, and use a VOC format dataset.
After training several epochs, the training is stuck and no error message is reported.

2020-03-03 09:01:13,561 atss_core.data.build WARNING: When using more than one image per GPU you may encounter an out-of-memory (OOM) error if your GPU does not have sufficient memory. If this happens, you can reduce SOLVER.IMS_PER_BATCH (for training) or TEST.IMS_PER_BATCH (for inference). For training, you must also adjust the learning rate and schedule length according to the linear scaling rule. See for example: https://github.com/facebookresearch/Detectron/blob/master/configs/getting_started/tutorial_1gpu_e2e_faster_rcnn_R-50-FPN.yaml#L14
2020-03-03 09:01:14,200 atss_core.trainer INFO: Start training
2020-03-03 09:01:41,085 atss_core.trainer INFO: eta: 11:11:34  iter: 20  loss: 2.3583 (2.4249)  loss_centerness: 0.6581 (0.6735)  loss_cls: 0.8432 (0.8624)  loss_reg: 0.8724 (0.8890)  time: 1.3581 (1.3440)  data: 0.2522 (0.3124)  lr: 0.000167  max mem: 4039
2020-03-03 09:02:08,343 atss_core.trainer INFO: eta: 11:15:50  iter: 40  loss: 2.1446 (2.2998)  loss_centerness: 0.6432 (0.6587)  loss_cls: 0.6304 (0.7604)  loss_reg: 0.8565 (0.8808)  time: 1.3038 (1.3535)  data: 0.2461 (0.2953)  lr: 0.000167  max mem: 4039
2020-03-03 09:02:37,059 atss_core.trainer INFO: eta: 11:29:04  iter: 60  loss: 2.1173 (2.2426)  loss_centerness: 0.6291 (0.6498)  loss_cls: 0.6318 (0.7219)  loss_reg: 0.8421 (0.8708)  time: 1.4707 (1.3809)  data: 0.3749 (0.3216)  lr: 0.000167  max mem: 4150
2020-03-03 09:03:05,780 atss_core.trainer INFO: eta: 11:35:29  iter: 80  loss: 2.0577 (2.1967)  loss_centerness: 0.6350 (0.6464)  loss_cls: 0.5637 (0.6818)  loss_reg: 0.8557 (0.8684)  time: 1.4177 (1.3947)  data: 0.3985 (0.3353)  lr: 0.000167  max mem: 4150
2020-03-03 09:03:34,258 atss_core.trainer INFO: eta: 11:37:56  iter: 100  loss: 2.0242 (2.1684)  loss_centerness: 0.6328 (0.6441)  loss_cls: 0.5646 (0.6626)  loss_reg: 0.8252 (0.8618)  time: 1.4187 (1.4005)  data: 0.2573 (0.3346)  lr: 0.000167  max mem: 4153
2020-03-03 09:04:01,555 atss_core.trainer INFO: eta: 11:34:30  iter: 120  loss: 2.0670 (2.1527)  loss_centerness: 0.6368 (0.6438)  loss_cls: 0.5709 (0.6459)  loss_reg: 0.8609 (0.8631)  time: 1.3416 (1.3946)  data: 0.2366 (0.3213)  lr: 0.000167  max mem: 4153
2020-03-03 09:04:28,856 atss_core.trainer INFO: eta: 11:31:56  iter: 140  loss: 1.9155 (2.1227)  loss_centerness: 0.6255 (0.6420)  loss_cls: 0.5054 (0.6274)  loss_reg: 0.7737 (0.8533)  time: 1.4071 (1.3904)  data: 0.2924 (0.3174)  lr: 0.000167  max mem: 4153
2020-03-03 09:04:57,257 atss_core.trainer INFO: eta: 11:33:19  iter: 160  loss: 1.9624 (2.1043)  loss_centerness: 0.6367 (0.6418)  loss_cls: 0.5202 (0.6145)  loss_reg: 0.8103 (0.8480)  time: 1.4475 (1.3941)  data: 0.2737 (0.3206)  lr: 0.000167  max mem: 4153

**Comment by kaikaizhu** [comment]
2020-03-03 09:01:07,404 atss_core INFO: Using 2 GPUs
2020-03-03 09:01:07,404 atss_core INFO: Namespace(config_file='configs/atss/atss_dcnv2_R_101_FPN_1x_water.yaml', distributed=True, local_rank=0, opts=['DATALOADER.NUM_WORKERS', '0', 'OUTPUT_DIR', 'training_dir/atss_dcnv2_R_101_FPN_1x/'], skip_test=False)
2020-03-03 09:01:07,405 atss_core INFO: Collecting env info (might take some time)
2020-03-03 09:01:11,510 atss_core INFO: 
PyTorch version: 1.1.0
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 14.04.4 LTS
GCC version: (GCC) 4.9.2
CMake version: Could not collect

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 9.0.176
GPU models and configuration: 
GPU 0: GeForce GTX 1080 Ti
GPU 1: GeForce GTX 1080 Ti
GPU 2: GeForce GTX 1080 Ti
GPU 3: GeForce GTX 1080 Ti
GPU 4: GeForce GTX 1080 Ti
GPU 5: GeForce GTX 1080 Ti
GPU 6: GeForce GTX 1080 Ti
GPU 7: GeForce GTX 1080 Ti

Nvidia driver version: 430.14
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.0.5
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so.5.1.5

Versions of relevant libraries:
[pip3] numpy==1.16.2
[pip3] torch==1.1.0
[pip3] torchfile==0.1.0
[pip3] torchvision==0.3.0
[conda] mkl                       11.3.3                        0    defaults
[conda] mkl-service               1.1.2                    py27_2    defaults
        Pillow (6.2.0)
2020-03-03 09:01:11,511 atss_core INFO: Loaded configuration file configs/atss/atss_dcnv2_R_101_FPN_1x_water.yaml
2020-03-03 09:01:11,511 atss_core INFO: 
MODEL:
  META_ARCHITECTURE: "GeneralizedRCNN"
  WEIGHT: "catalog://ImageNetPretrained/MSRA/R-101"
  RPN_ONLY: True
  ATSS_ON: True
  BACKBONE:
    CONV_BODY: "R-101-FPN-RETINANET"
  RESNETS:
    BACKBONE_OUT_CHANNELS: 256
    STAGE_WITH_DCN: (False, True, True, True)
    WITH_MODULATED_DCN: True
    DEFORMABLE_GROUPS: 1
  RETINANET:
    USE_C5: False
  ATSS:
    ANCHOR_SIZES: (64, 128, 256, 512, 1024) # 8S
    ASPECT_RATIOS: (1.0,)
    SCALES_PER_OCTAVE: 1
    USE_DCN_IN_TOWER: True
    POSITIVE_TYPE: 'ATSS' # how to select positves: ATSS (Ours) , SSC (FCOS), IoU (RetinaNet)
    TOPK: 9 # topk for selecting candidate positive samples from each level
    REGRESSION_TYPE: 'BOX' # regressing from a 'BOX' or a 'POINT'
DATASETS:
  TRAIN: ("voc_water_train",)
  TEST: ("coco_2017_val",)
INPUT:
  MIN_SIZE_TRAIN: (800,)
  MAX_SIZE_TRAIN: 1333
  MIN_SIZE_TEST: 800
  MAX_SIZE_TEST: 1333
DATALOADER:
  SIZE_DIVISIBILITY: 32
SOLVER:
  # BASE_LR: 0.0025
  # STEPS: [0, 30000, 40000]
  # MAX_ITER: 60000
  BASE_LR: 0.005
  STEPS: [0, 15000, 20000]
  MAX_ITER: 30000
  WEIGHT_DECAY: 0.0001
  IMS_PER_BATCH: 4
  WARMUP_METHOD: "constant"

**Comment by kaikaizhu** [comment]
2020-03-03 09:01:07,404 atss_core INFO: Using 2 GPUs
2020-03-03 09:01:07,404 atss_core INFO: Namespace(config_file='configs/atss/atss_dcnv2_R_101_FPN_1x_water.yaml', distributed=True, local_rank=0, opts=['DATALOADER.NUM_WORKERS', '0', 'OUTPUT_DIR', 'training_dir/atss_dcnv2_R_101_FPN_1x/'], skip_test=False)
2020-03-03 09:01:07,405 atss_core INFO: Collecting env info (might take some time)
2020-03-03 09:01:11,510 atss_core INFO: 
PyTorch version: 1.1.0
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 14.04.4 LTS
GCC version: (GCC) 4.9.2
CMake version: Could not collect

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 9.0.176
GPU models and configuration: 
GPU 0: GeForce GTX 1080 Ti
GPU 1: GeForce GTX 1080 Ti
GPU 2: GeForce GTX 1080 Ti
GPU 3: GeForce GTX 1080 Ti
GPU 4: GeForce GTX 1080 Ti
GPU 5: GeForce GTX 1080 Ti
GPU 6: GeForce GTX 1080 Ti
GPU 7: GeForce GTX 1080 Ti

Nvidia driver version: 430.14
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.0.5
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so.5.1.5

Versions of relevant libraries:
[pip3] numpy==1.16.2
[pip3] torch==1.1.0
[pip3] torchfile==0.1.0
[pip3] torchvision==0.3.0
[conda] mkl                       11.3.3                        0    defaults
[conda] mkl-service               1.1.2                    py27_2    defaults
        Pillow (6.2.0)
2020-03-03 09:01:11,511 atss_core INFO: Loaded configuration file configs/atss/atss_dcnv2_R_101_FPN_1x_water.yaml
2020-03-03 09:01:11,511 atss_core INFO: 
MODEL:
  META_ARCHITECTURE: "GeneralizedRCNN"
  WEIGHT: "catalog://ImageNetPretrained/MSRA/R-101"
  RPN_ONLY: True
  ATSS_ON: True
  BACKBONE:
    CONV_BODY: "R-101-FPN-RETINANET"
  RESNETS:
    BACKBONE_OUT_CHANNELS: 256
    STAGE_WITH_DCN: (False, True, True, True)
    WITH_MODULATED_DCN: True
    DEFORMABLE_GROUPS: 1
  RETINANET:
    USE_C5: False
  ATSS:
    ANCHOR_SIZES: (64, 128, 256, 512, 1024) # 8S
    ASPECT_RATIOS: (1.0,)
    SCALES_PER_OCTAVE: 1
    USE_DCN_IN_TOWER: True
    POSITIVE_TYPE: 'ATSS' # how to select positves: ATSS (Ours) , SSC (FCOS), IoU (RetinaNet)
    TOPK: 9 # topk for selecting candidate positive samples from each level
    REGRESSION_TYPE: 'BOX' # regressing from a 'BOX' or a 'POINT'
DATASETS:
  TRAIN: ("voc_water_train",)
  TEST: ("coco_2017_val",)
INPUT:
  MIN_SIZE_TRAIN: (800,)
  MAX_SIZE_TRAIN: 1333
  MIN_SIZE_TEST: 800
  MAX_SIZE_TEST: 1333
DATALOADER:
  SIZE_DIVISIBILITY: 32
SOLVER:
  BASE_LR: 0.005
  STEPS: [0, 15000, 20000]
  MAX_ITER: 30000
  WEIGHT_DECAY: 0.0001
  IMS_PER_BATCH: 4
  WARMUP_METHOD: "constant"

**Comment by sfzhang15** [comment]
@kaikaizhu 
Sorry for the late response. Does your own dataset have empty or wrongly-annotated images?
